awx/main/scheduler/task_manager.py

```diff
@@ -214,13 +215,16 @@ def start_task(self, task, rampart_group, dependent_tasks=None, instance=None):
         dependencies = [{'type': get_type_for_model(type(t)), 'id': t.id} for t in dependent_tasks]
 
         controller_node = None
+        container_controller = None
         if task.supports_isolation() and rampart_group.controller_id:
             try:
                 controller_node = rampart_group.choose_online_controller_node()
             except IndexError:
                 logger.debug("No controllers available in group {} to run {}".format(
                              rampart_group.name, task.log_format))
                 return
+        elif rampart_group and rampart_group.is_containerized:
+            container_controller = random.choice(task.global_instance_groups)
```

Comment:
*sigh* a problem we still need to solve .. compute consumption on the controller side.

Reply:
yeeep

Reply:
Noodling over the usage of `global_instance_groups` here. This requires users to make sure the special `tower` group exists and to add nodes to in order for this to succeed.  But can we do better than this?

We know that `rampart_group` is a containerized group (it has a kubernetes type cred assigned to it). So should all the `instance`(s) belonging to that special group be considered controllers for that group and we chose from that list?